list.go

```diff
@@ -401,3 +404,127 @@ func (db *RoseDB) listSequence(headSeq, tailSeq uint32, index int) (uint32, erro
 	}
 	return seq, nil
 }
+
+// LRem removes the first count occurrences of elements equal to element from the list stored at key.
+// The count argument influences the operation in the following ways:
+// count > 0: Remove elements equal to element moving from head to tail.
+// count < 0: Remove elements equal to element moving from tail to head.
+// count = 0: Remove all elements equal to element.
+// Note that key does not exist, the command will always return 0.
+func (db *RoseDB) LRem(key []byte, count int, value []byte) (int, error) {
+	db.listIndex.mu.Lock()
+	defer db.listIndex.mu.Unlock()
+
+	if count == 0 {
+		count = math.MaxInt
+	}
+	var discardCount int
+	idxTree := db.listIndex.trees[string(key)]
+	if idxTree == nil {
+		return discardCount, nil
+	}
+	// get List DataType meta info
+	headSeq, tailSeq, err := db.listMeta(idxTree, key)
+	if err != nil {
+		return discardCount, err
+	}
+
+	reserveSeq, discardSeq, reserveValueSeq := make([]uint32, 0), make([]uint32, 0), make([][]byte, 0)
+	if count > 0 {
```

Comment:
This part is not very clear. I think the entire process is:
1. get all values of the key
2. iterate and find a value equal to the given one
3. filter and rewrite values

```
values = db.LRange(key, 0, -1)
valuesIds = []
for i, val range values {
    if val == value {
        valuesIds = append(valuesIds, i)
    }
}

if len(valuesIds) == 0 {
    return
}

// delete the ids(should be removed) from valueIds according to param count.
valuesIds = valueIds[3:] // just an example

// then the valuesIds array are the values we should rewrite.
valuesMap = {}
for i range valueIds {
    valuesMap[i] = struct{}{}
}

// reset meta

// iterate and rewrite
for i, val range values {
    if val == value && i not existed in valuesMap {
        contine
    }
    pushInternal()
}
```